reference/tools: add FAQs and update mydumper doc (#1728)
pingcap/docs: dev/reference/tools/mydumper.md

### Usage example

To back up data from TiDB, add command line parameters to the following command according to your situation:

{{< copyable "shell-regular" >}}

```bash
./bin/mydumper -h 127.0.0.1 -u root -P 4000
```

## FAQ

### How to determine if the Mydumper I am using is the PingCAP optimized version?

Execute the following command：

{{< copyable "shell-regular" >}}

```bash
./bin/mydumper -V
```

If the output contains `git_hash` (`d3e6fec8b069daee772d0dbaa47579f67a5947e7` in the following example), you are using the PingCAP optimized version of Mydumper:

```
mydumper 0.9.5 (d3e6fec8b069daee772d0dbaa47579f67a5947e7), built against MySQL 5.7.24
```

> ### What can I do if an "invalid mydumper files for there are no `-schema-create.sql` files found" error is reported when using Loader to restore the data backed up by Mydumper?

Reviewer: ```suggestion
### How to resolve the "invalid mydumper files for there are no `-schema-create.sql` files found" error when using Loader to restore the data backed up by Mydumper?
```

### Why is the TIMESTAMP type data exported using Mydumper inconsistent with that in the database?

Check whether the time zone of the server running Mydumper is consistent with that of the database. Mydumper converts the TIMESTAMP type data according to the time zone of the server. You can add the `--skip-tz-utc` option to disable the conversion of dates and times.

### How to configure the `-F,--chunk-filesize` option of Mydumper?

Mydumper divides the data of each table into multiple `chunk` according to the value of this option during backup. Each `chunk` is saved to a file with a size of about `chunk-filesize`. In this way, the data is divided into multiple files so that the import speed is improved by using the parallel processing of Loader/TiDB Lightning. If you later use **Loader** to restore the backup files, it is recommended to set the value of this option to `64` (in MB); If you use **TiDB Lightning** to restore files, `256` (in MB) is recommended.

### How to configure the `-s --statement-size` option of Mydumper?

> Mydumper uses this option to control the size of `Insert Statement` which defaults to `10000000` (about 1 MB). Use this option to avoid the following errors occurring when restoring data:

Reviewer: ```suggestion
Mydumper uses this option to control the size of `Insert Statement` which defaults to `10000000` (about 1 MB). Use this option to avoid the following error when restoring data:
```

```log
packet for query is too large. Try adjusting the 'max_allowed_packet' variable
```

The default value meets the requirements in most cases, but **if it is a wide table, the size of a single row of data might exceed the limit of `statement-size`, and Mydumper reports the following warning**:

```log
Row bigger than statement_size for xxx
```

If you restore the data in this situation, Mydumper still reports the `packet for query is too large` error. To solve this problem, modify the following two configurations (take `128 MB` as an example):

* Execute `set @@global.max_allowed_packet=134217728` (`134217728` = `128 MB`) in TiDB server.
* Add the `max-allowed-packet=128M` line to the DB configuration of Loader or DM task's configuration file according to your situation. Then, restart the process or task.

### How to set the `-l, --long-query-guard` option of Mydumper?

> Set the value of this option to the estimated time required for backup. If Mydumper runs longer than the value of this option, it reports an error and exits. It is recommended to set the value to `7200` (in seconds) for the first time and then modify it according to your actual backup time.

Reviewer: ```suggestion
Set the value of this option to the estimated time required for the backup. If Mydumper runs longer than this value, it reports an error and exits. It is recommended to set the value to `7200` (in seconds) for the first time of your backup and then modify it according to your actual backup time.
```

### How to set the `--tidb-force-priority` option of Mydumper?

This option can only be set when backing up TiDB’s data. It can be set to `LOW_PRIORITY`, `DELAYED`, or `HIGH_PRIORITY`. If you do not want data backup to affect online services, it is recommended to set this option to `LOW_PRIORITY`; if the backup has a higher priority, `HIGH_PRIORITY` is recommended.

> ### What can I do if an "GC life time is short than transaction duration" error is reported when using Mydumper to back up TiDB's data?

Reviewer: ```suggestion
### How to resolve the "GC life time is short than transaction duration" error when using Mydumper to back up TiDB's data?
```

    ```

    {{< copyable "sql" >}}

    ```sql
    update mysql.tidb set VARIABLE_VALUE = '720h' where VARIABLE_NAME = 'tikv_gc_life_time';
    ```

2. Set the value of `tikv_gc_life_time` to the initial one after the backup is complete:

    {{< copyable "sql" >}}

    ```sql
    update mysql.tidb set VARIABLE_VALUE = '10m0s' where VARIABLE_NAME = 'tikv_gc_life_time';
    ```

### Do I need to configure the `--tidb-rowid` option of Mydumper?

If this option is set to true, the exported data contains the data of TiDB's hidden columns. Using hidden columns when restoring data to TiDB might cause data inconsistency. Currently, it is not recommended to use this option.

### What can I do if Mydumper reports an "Segmentation Fault" error?

This bug has been fixed. If the error persists, you can upgrade to the latest version of Mydumper.

### What can I do if Mydumper reports an "Error dumping table ({schema}.{table}) data: line ...... (total length ...)" error?

This error occurs when Mydumper parses SQL statements. In this situation, use the latest version of Mydumper. If this error persists, you can file an issue to [mydumper/issues](https://github.com/pingcap/mydumper/issues).

> ### What can I do if Mydumper reports an "Failed to set tidb_snapshot: parsing time \"20190901-10:15:00 +0800\" as \"20190901-10:15:00 +0700 MST\": cannot parse \"\" as \"MST\"" error?

Reviewer: ```suggestion
### What can I do if Mydumper reports an `"Failed to set tidb_snapshot: parsing time \"20190901-10:15:00 +0800\" as \"20190901-10:15:00 +0700 MST\": cannot parse \"\" as \"MST\""` error?
```
